src/bp-templates/bp-nouveau/includes/activity/template-tags.php

```diff
@@ -272,21 +273,127 @@ function bp_nouveau_activity_state() {
 			</span>
 		</a>
 		<span class="ac-state-separator">&middot;</span>
-		<a href="#" class="activity-state-comments">
-			<span class="comments-count">
-				<?php
-				if ( $comment_count > 1 ) {
-					echo $comment_count . ' ' . __( 'Comments', 'buddyboss' );
-				} else {
-					echo $comment_count . ' ' . __( 'Comment', 'buddyboss' );
-				}
-				?>
-			</span>
-		</a>
+		<?php if ( bp_activity_can_comment() ) : ?>
+			<a href="#" class="activity-state-comments">
+				<span class="comments-count">
+					<?php
+					if ( $comment_count > 1 ) {
+						echo $comment_count . ' ' . __( 'Comments', 'buddyboss' );
+					} else {
+						echo $comment_count . ' ' . __( 'Comment', 'buddyboss' );
+					}
+					?>
+				</span>
+			</a>
+		<?php endif; ?>
+
+		<?php if ( ! bp_activity_can_comment() ) : ?>
+			<div href="#" class="activity-state-comments">
+				<span class="comments-count">
+					<?php
+					if ( $comment_count > 1 ) {
+						echo $comment_count . ' ' . __( 'Comments', 'buddyboss' );
+					} else {
+						echo $comment_count . ' ' . __( 'Comment', 'buddyboss' );
+					}
+					?>
+				</span>
+			</div>
+		<?php endif; ?>
```

Comment:
@asaquzzamanbb I am seeing the same code in both the condition then why we need the condition and need duplication code?

Reply:
@chetansatasiya 
first condition contain <a> tag wrap and second condition <div> tag warp. But i need to remove the href="#" from div tag

Reply:
@chetansatasiya 
Remove the `href="#"`
https://github.com/buddyboss/buddyboss-platform/pull/2290/commits/248bc2a6c745768a954e0db8da5d028f3d4a6eb4